# Que - 1

In [1]:
import pandas as pd 
import numpy as np
from scipy import stats
from sklearn import datasets
import statsmodels.api as sm
from scipy.stats import chi2_contingency

In [2]:
data= pd.read_csv("Cutlets.csv")
data.head()

UnitA   UnitB
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [3]:
data.isnull().sum()

UnitA    0
UnitB    0
dtype: int64

In [4]:
data.describe()

UnitA      UnitB
count  35.000000  35.000000
mean    7.019091   6.964297
std     0.288408   0.343401
min     6.437600   6.038000
25%     6.831500   6.753600
50%     6.943800   6.939900
75%     7.280550   7.195000
max     7.516900   7.545900

In [5]:
data.columns

Index(['UnitA', 'UnitB'], dtype='object')

In [6]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35 entries, 0 to 34
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   UnitA   35 non-null     float64
 1   UnitB   35 non-null     float64
dtypes: float64(2)
memory usage: 688.0 bytes


In [7]:
stats.ttest_ind(data.UnitA, data.UnitB)

Ttest_indResult(statistic=0.7228688704678061, pvalue=0.4722394724599501)

* Ans --> as the obtained  p value > alpha value i.e. 0.4722 > 0.05, we can accept Null hypothesis. As we observed there is no significant difference in the diameter of the cutlet between 2 units.

# Que - 2

In [8]:
hop = pd.read_csv("LabTAT.csv")

In [9]:
hop.head()

Laboratory1  Laboratory2  Laboratory3  Laboratory4
0       185.35       165.53       176.70       166.13
1       170.49       185.91       198.45       160.79
2       192.77       194.92       201.23       185.18
3       177.33       183.00       199.61       176.42
4       193.41       169.57       204.63       152.60

In [10]:
hop.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 120 entries, 0 to 119
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Laboratory1  120 non-null    float64
 1   Laboratory2  120 non-null    float64
 2   Laboratory3  120 non-null    float64
 3   Laboratory4  120 non-null    float64
dtypes: float64(4)
memory usage: 3.9 KB


In [11]:
hop.describe()

Laboratory1  Laboratory2  Laboratory3  Laboratory4
count   120.000000   120.000000   120.000000    120.00000
mean    178.361583   178.902917   199.913250    163.68275
std      13.173594    14.957114    16.539033     15.08508
min     138.300000   140.550000   159.690000    124.06000
25%     170.335000   168.025000   188.232500    154.05000
50%     178.530000   178.870000   199.805000    164.42500
75%     186.535000   189.112500   211.332500    172.88250
max     216.390000   217.860000   238.700000    205.18000

In [12]:
hop.shape

(120, 4)

In [13]:
hop.isnull().sum()

Laboratory1    0
Laboratory2    0
Laboratory3    0
Laboratory4    0
dtype: int64

In [14]:
from sklearn import datasets

In [15]:
# one way anova test using statsmodels module
stats.f_oneway(hop["Laboratory1"],hop["Laboratory2"],hop["Laboratory3"],hop["Laboratory4"])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

In [16]:
stats.f_oneway(hop.iloc[:,0],hop.iloc[:,1],hop.iloc[:,2],hop.iloc[:,3])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

* p-value <0.05. Hence, we can reject the Null Hypothesis – there are no differences among different groups
P value=2.1156708949992414e-57,p<0.05,P low , fail to accept null hypothesis , what alternate hypothesis says "Average turn around time of all 4 laboratories are not same"

# Que - 3

In [17]:
que3 = pd.read_csv("BuyerRatio.csv")

In [18]:
que3

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [19]:
stats.chi2_contingency([que3['East'],que3['West'],que3['North'],que3['South']])

(1.5959455386610577,
 0.6603094907091882,
 3,
 array([[  42.76531299,  442.23468701],
        [ 146.81287862, 1518.18712138],
        [ 131.11756787, 1355.88243213],
        [  72.30424052,  747.69575948]]))

* Here from the above data if u observe P-value>0.05 we Accept Null Hypothesis Proportion Of buyersratio Across all the Regions Should Be Same Statistically Proved

### Que 4

In [20]:
que4 = pd.read_csv('Costomer+OrderForm.csv')
que4

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [21]:
data = que4.apply(pd.value_counts)
data

Phillippines  Indonesia  Malta  India
Error Free           271        267    269    280
Defective             29         33     31     20

In [22]:
stats.chi2_contingency(data)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

* p>0.05 we accept null hypothesis, that there is a defect varies by a center and the Tukeyhsd states that there is a significant different among the groups